src/internal/m365/collection/site/mock/list.go

```diff
@@ -94,21 +95,72 @@ func (lh *ListHandler) Check(t *testing.T, expected []string) {
 }
 
 type ListRestoreHandler struct {
-	List models.Listable
-	Err  error
+	ignoreSubsequentPostListFails bool
```

Comment:
this feels a little cumbersome.  something we've done in other handler mocks is to have props like `idx int, errs []error`.  Then if the test expects two calls using the mock, they can add `errs: [err, nil]` (or just [err], and let out of bounds always return nil).  That tends to be a bit more straight forward for handling.